### ISSUE TITLE
Multiomics - Server Question

### ISSUE THREAD

**Issue by sstemann** [NCATSTranslator/TranslatorPM#60]
We're trying to document our federated dev environment, I can't seem to get the server info from SmartAPI, could you take a look at the table here:  

https://github.com/NCATSTranslator/TranslatorPM/blob/main/Translator%20Server%20Info.md

And either update directly in the markdown or update this ticket.  We have the following KPs for your team:
- Wellness Provider
- BigGim
- Drug Response Provider
- TCGA Mutation Frequency Provider
- Clinical Risk Provider
- BigQuery Table

@NCATSTranslator/multiomics-provider sorry to tag the whole team but I'm not sure who to work with to get these updates!  Thanks for your help!

**Comment by rtroper** [comment]
Okay, I've created a pull request to add the server for our KPs that are hosted on the service provider server: https://github.com/NCATSTranslator/TranslatorPM/pull/67

**Comment by sstemann** [comment]
@rtroper thank you!  for the goal to have a hosted Dev environment - will any of the Multiomics KPs be hosted by NCATS or do you prefer to engage in an SLA?

**Comment by sstemann** [comment]
@rtroper just following up on the comment above and adding one more question - is Big Query Table a KP?  thanks!

**Comment by rtroper** [comment]
> @rtroper just following up on the comment above and adding one more question - is Big Query Table a KP? thanks!

Sarah - We do not plan to have any KPs hosted in BigQuery. There's a chance that, for at least one KP, we'll store the data for the KP in a BigQuery table and then use a parser in combination with the Service Provider group's Biothings Studio tool to load (or update) the KP hosted on their server. Our plan right now is to keep all our KPs on their server (biothings.ncats.io).

**Comment by rtroper** [comment]
> @rtroper thank you! for the goal to have a hosted Dev environment - will any of the Multiomics KPs be hosted by NCATS or do you prefer to engage in an SLA?

I apologize I haven't responded to this yet. Let me get back to you shortly.

**Comment by sstemann** [comment]
thanks!

**Comment by kevinxin90** [comment]
Hi @sstemann , so currently all KPs from Multiomics Provider are hosted by Service Provider at `biothings.ncats.io`. This is an AWS instance assigned to the Service Provider by NCATS. Does that count as **"hosted by NCATS"**. Or do we still need a SLA?

**Comment by sstemann** [comment]
thank you for bringing this up - so I know Ryan's been saying this but i didn't click for me.  So, when there's a new version of a multiomics KP, you push it to the biothings instance?

**Comment by kevinxin90** [comment]
Yes. That's right. There's a semi-automated pipeline to go from a KGX file to an API served at the biothings instance.

**Comment by sstemann** [comment]
Seems okay to me, I know currently Biothings is hosted by an NCATS Cloud instance, but this is not an NCATS ITRB instance and if you would like to continue NCATS to host, you'll need to utilize the NCATS ITRB Pipeline.  Is that okay for biothings?

If so, then biothings will take on the responsibility of moving the KGX files through the NCATS ITRB pipeline?  

Are there other KPs that you can identify will also be part of this process?  Thanks!

**Comment by kevinxin90** [comment]
@sstemann Hi Sarah, are there documentations which we can learn about NCATS ITRB Pipeline, so we can evaluate the feasibility?

Or maybe @newgene already have plans for that?

**Comment by newgene** [comment]
I also don't know about ITRB Pipeline. @sstemann we are happy to continue hosting BioThings-based KPs (for Service Provider, Multiomics Provider, Textmining Provider, and other teams) using NCATS cloud instances, so please advise what we need to work through the process. Thanks.

**Comment by sstemann** [comment]
Sure – we’re just starting the process (and I thought SmartAPI went through it with N3C.  For now, we are targeting ITRB hosting dev by May 1 which supports continuous deployments,  my understanding is that prod will not have CD but more of a staged/requested pipeline.

https://docs.google.com/document/d/1DEHrRH-euvYCO8PrF4eyQjlOVp63mRFx99YvKYIjGR8/edit?usp=sharing

The resources currently provided on this program have been NCATS but not part of ITRB, and may be sunsetted once everything is either ITRB or self-hosted with SLAs in place.  https://docs.google.com/document/d/1DEHrRH-euvYCO8PrF4eyQjlOVp63mRFx99YvKYIjGR8/edit?usp=sharing

**Comment by newgene** [comment]
@sstemann awesome! Thanks. You are right, our team did have done this process for other projects relevant to N3C and CD2H. It just did not ring the bell when I heard the term "ITRB pipeline" :-)

We will plan it accordingly for our translator projects then. And since the BioThings stack from Service Provider provides the API hosting for other translator teams too. This effort should help reduce or even eliminate the overhead for other teams too.

**Comment by kevinxin90** [comment]
@sstemann Hi Sarah, @newgene and I would also like to explore the option of go with SLA. Could you share with us the agreement please? Thank you！
